docs/source/tracking/data-api.rst

```diff
@@ -0,0 +1,227 @@
+
+   # Load the data again
+   loaded_data = pd.read_csv(local_dataset, delimiter=CSV_DELIMITER)
+
+The print statement from above resolves to the local file that was created when calling ``load``.
+
+.. code-block:: shell
+
+    The local file where the data has been downloaded to:
+    /var/folders/cd/n8n0rm2x53l_s0xv_j_xklb00000gp/T/tmpuxwtrul1/winequality-white.csv
+
+What is the purpose of ``mlflow.data``?
+---------------------------------------
+
+The ``mlflow.data`` module serves the crucial role of associating datasets with MLflow runs. Aside from the obvious utility of having a record 
+associated with an MLflow run to the dataset that was used during training, there are some integrations within MLflow that allow for direct 
+usage of Datasets that have been logged with the :py:func:`mlflow.log_input` API. 
+
+How to use a Dataset with MLflow evaluate
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. note:: 
+    The integration of Datasets with MLflow evaluate was introduced in MLflow 2.8.0. Previous versions do not have this functionality.
+
+To see how this integration functions, let's take a look at a fairly simple and typical classification task. 
+
+.. code-block:: python
+
+    import pandas as pd
+
+    from sklearn.model_selection import train_test_split
+    from sklearn.preprocessing import LabelEncoder
+    import xgboost
+
+    import mlflow
+    from mlflow.data.pandas_dataset import PandasDataset
+
+    CSV_DELIMITER = ";"
+
+    dataset_source_url = "https://raw.githubusercontent.com/mlflow/mlflow/master/tests/datasets/winequality-white.csv"
+    raw_data = pd.read_csv(dataset_source_url, delimiter=CSV_DELIMITER)
+
+    # Extract the features and target data separately
+    y = raw_data["quality"]
+    X = raw_data.drop("quality", axis=1)
+
+    # Split the data into training and test sets
+    X_train, X_test, y_train, y_test = train_test_split(
+        X, y, test_size=0.33, random_state=17
+    )
+
+    # Create a label encoder object
+    le = LabelEncoder()
+
+    # Fit and transform the target variable
+    y_train_encoded = le.fit_transform(y_train)
+    y_test_encoded = le.transform(y_test)
+
+    # Fit an XGBoost binary classifier on the training data split
+    model = xgboost.XGBClassifier().fit(X_train, y_train_encoded)
+
+    # Build the Evaluation Dataset from the test set
+    y_test_pred = model.predict(X=X_test)
+
+    eval_data = X_test
+    eval_data["label"] = y_test
+
+    # Assign the decoded predictions to the Evaluation Dataset
+    eval_data["predictions"] = le.inverse_transform(y_test_pred)
+
+    # Create the PandasDataset for use in mlflow evaluate
+    pd_dataset = mlflow.data.from_pandas(
+        eval_data, predictions="predictions", targets="label"
+    )
+
+    mlflow.set_experiment("White Wine Quality")
+
+    # Log the Dataset, model, and execute an evaluation run using the configured Dataset
+    with mlflow.start_run() as run:
+        mlflow.log_input(pd_dataset, context="training")
```

Comment:
When we use the dataset for evaluation, we don't need to call `log_input()` but MLflow automatically logs it, which I think is very cool. Perhaps showing verbose `log_input` make it look less convenient?

Reply:
Good point :) removed and added a note explaining how we're calling this on behalf of the users